### PR TITLE
Implement delete edges

### PR DESCRIPTION
**Pull request by zlcook**
Issue #172 
Delete Sentence: 
* `DELETE EDGE edge_type(<vertex_id>-><vertex_id>[@\<weight>] [, <vertex_id>-><vertex_id>[@\<weight>]]*)

### CODE REVIEW

**Comment by whitewum** [comment]
should this pr rebase to master? @zlcook

File: src/graph/test/DeleteEdgeTest.cpp [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT VERTEX person(name, age) VALUES 1001:(\"Zhangsan\", 22),"
+                          "1002:(\"Lisi\", 23), 1003:(\"Jack\", 18), 1004:(\"Rose\", 19)";
+        auto code = client_->execute(cmd, resp);
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, code);
+    }
+    {
+        cpp2::ExecutionResponse resp;
+        std::string cmd = "INSERT EDGE friend(intimacy) VALUES 1001->1002@15:(90),"
```
Comment (whitewum) [inline src/graph/test/DeleteEdgeTest.cpp]: add two (different types or different ranks) edgess with the same src and the same dst. Then delete one . 

and try to double delete the same edge.

delete a not exist edge. 

check: both the in and out keys are not exists in storage ( not partially deleted)

Comment (zlcook) [reply src/graph/test/DeleteEdgeTest.cpp]: Ok

**Comment by zlcook** [comment]
> I think current storage client can't guarantee that both in-key and out-key can be deleted. Since they are in different shard on different machine. Would you add a comment to address this?

You are right, I will add a comment.

> BTW. What about the feature to delete a vertex (with all the versions)?
 
* Basic idea:
graph layer need send twice requests to complete it.  
1. First request, delete the vertex and it's out-edges( they located in same partition), then construct `EdgeKeys` of in-edges (located in different partitions ) according to out-edges and return to graph layer.
2. Second request, graph layer delete the in-edges from the first request.

In order to ensure the in-edges can eventually be by removed, we can record the `vertex` needed to be deleted  in `meta server`.  If the `second request` failed, we can continue to delete in-edges by compaction.

**Comment by whitewum** [comment]
> > I think current storage client can't guarantee that both in-key and out-key can be deleted. Since they are in different shard on different machine. Would you add a comment to address this?
> 
> You are right, I will add a comment.
> 
> > BTW. What about the feature to delete a vertex (with all the versions)?
> 
> * Basic idea:
>   graph layer need send twice requests to complete it.
> 
> 1. First request, delete the vertex and it's out-edges( they located in same partition), then construct `EdgeKeys` of in-edges (located in different partitions ) according to out-edges and return to graph layer.
> 2. Second request, graph layer delete the in-edges coming from the first request.
> 
> In order to ensure the in-edges can eventually be removed, we can record the `vertex` needed to be deleted in `meta server`. If the `second request` failed, we can continue to delete in-edges by compaction.

@dutor @dangleptr what do you think about delete a vertex with all the edges.  In GB, it allows dangling edges.

**Comment by dangleptr** [comment]
Ready to review now?

**Pull request closed without merging**